app/screens/search/search.js

```diff
@@ -92,7 +92,7 @@ class Search extends Component {
     componentDidUpdate() {
         const {searchingStatus, recent} = this.props;
         const recentLenght = recent.length;
-        const shouldScroll = searchingStatus === RequestStatus.SUCCESS || searchingStatus === RequestStatus.STARTED;
+        const shouldScroll = (searchingStatus === RequestStatus.SUCCESS || searchingStatus === RequestStatus.STARTED) && !this.recentRemoved;
```

Comment:
It actually does receive the previous props and state (and I think context as well, but we don't use that anywhere). We use it all over the web app